main.c

```diff
@@ -490,3 +492,13 @@ int main(void)
 	run_aseba_main_loop();
 }
 
+
+
+void AsebaVMResetCB(AsebaVMState *vm) {
+	memset(vm->variables, 0, vm->variablesSize*sizeof(sint16));
+	vmVariables.id = vmState.nodeId;
+	vmVariables.productid = PRODUCT_ID;
+	vmVariables.fwversion[0] = FW_VERSION;
+	vmVariables.fwversion[1] = FW_VARIANT;
+	soft_reset(vm);
+}
```

Comment:
You should first reset the internal robot state (such as the network) and then memset.
Otherwise there is a race between the memset and the network disable which could lead to non-reseted variables. (ie, you could have a prox interrupt between the memset() and the prox_network_disable(), which will set some variables).